segmentstore/server/src/test/java/io/pravega/segmentstore/server/writer/AckCalculatorTests.java

```diff
@@ -94,6 +95,36 @@ public void testGetHighestCommittedSequenceNumber() {
         Assert.assertEquals("Unexpected result for Set with partial values when LRSN is infinite.", expectedResult, result);
     }
 
+    /**
+     * Tests the {@link AckCalculator#getLowestUncommittedSequenceNumber} method.
+     */
+    @Test
+    public void testGetLowestUncommittedSequenceNumber() {
+        final int processorCount = 2;
+        WriterState state = new WriterState();
+        AckCalculator calc = new AckCalculator(state);
+
+        ArrayList<TestProcessor> processors = new ArrayList<>();
+        for (int i = 0; i < processorCount; i++) {
+            processors.add(new TestProcessor());
+        }
+
+        // Everything up-to-date.
+        processors.forEach(p -> p.setLowestUncommittedSequenceNumber(Operation.NO_SEQUENCE_NUMBER));
+        long lusn = calc.getLowestUncommittedSequenceNumber(processors);
+        Assert.assertEquals("Unexpected result when all processors up-to-date.", Operation.NO_SEQUENCE_NUMBER, lusn);
+
+        // One processor not up-to-date.
```

Comment:
Without the change of this PR, is it right that this case and the one starting in line 122 would fail?

Reply:
No, just this one would have failed. The previous code worked correctly for checks at line 112 and 122.